Add missing closing `</li>` tags

andrasguseo opened the pull request:
[TEC-4974]

* The closing </li> tags are missing for the items in the Table on Contents on the General and Display settings pages.
* Added 2 filters to allow filtering the TOC contents. This can be helpful when adding settings sections, for example for an extension.

[TEC-4974]: https://stellarwp.atlassian.net/browse/TEC-4974?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

andrasguseo commented:
@bordoni This is an older one. Main probably needs to be pulled in for an update.

bordoni commented on src/admin-views/tribe-options-general.php:
```diff
@@ -127,52 +127,59 @@
+ *
+ * @var array $tec_events_display_toc Array of items of the TOC.
+ */
+$tec_events_general_toc = apply_filters( 'tec_events_general_settings_toc', $tec_events_general_toc );
```
The docblock needs a `@since` and use `@param` folr filters, no need to filter into the variable, than apply the addition. Do it all in one line.

bordoni commented on src/admin-views/tribe-options-display.php:
```diff
@@ -42,66 +43,72 @@
+ *
+ * @var array $tec_events_display_toc Array of items of the TOC.
+ */
+$tec_events_display_toc = apply_filters( 'tec_events_display_settings_toc', $tec_events_display_toc );
```
The docblock needs a `@since` and use `@param` folr filters, no need to filter into the variable, than apply the addition. Do it all in one line.

bordoni reviewed (changes requested):
Lets avoid change formatting on the whole file, make the changes only on the `<li>` and to the filter.

Move the spacing changes to a separate PR.

Also would love to revisit that actual filter names to be a bit more verbose

andrasguseo commented:
@bordoni I did an interactive rebase (with the help of chatGPT) to remove the code formatting commits. If there's a better way to do it, I'd love to know.
Addressed your comments.
Also happy to talk about hook naming conventions.

bordoni commented on src/admin-views/tribe-options-general.php:
```diff
@@ -173,7 +173,14 @@
+ *
+ * @since TBD
+ *
+ * @var array $tec_events_display_toc Array of items of the TOC.
```
```suggestion
 * @param array $tec_events_display_toc Array of items of the TOC.
```

bordoni commented on src/admin-views/tribe-options-display.php:
```diff
@@ -101,7 +100,14 @@
+ *
+ * @since TBD
+ *
+ * @var array $tec_events_display_toc Array of items of the TOC.
```
```suggestion
 * @param array $tec_events_display_toc Array of items of the TOC.
```

bordoni reviewed (changes requested):
A couple of things:
* Update the title of the PR to be more descriptive
* Release branch needs to be `release/T24.asag`
* Make sure we are not commits changes to `common` submodule.

andrasguseo commented:
@bordoni 

> Make sure we are not commits changes to common submodule.

Not sure how common files got in the mix. I'd like to know how this could have happened and how I can avoid it in the future. Also, I need help removing them. Thx.
